### DISCUSSION TITLE
A question of personal armour

### DISCUSSION THREAD

**Discussion by ultrak2k** [tsunamayo/Starship-EVO#4189] [Suggestions]
Well I was thinking recently:
-We have handheld weapons, and we have really small bricks, and handheld weapons downscale bricks

I'm curious - would it be possible to add brick-buildable body armour using this tech? either freebuilt somehow, or having some baseline suit you can overlay bricks on somehow?

Just a quick question really

**Comment by cloudrambler** [comment]
A thought about this idea is the complexity of how that armour would interact with ship interiors. When characters enter/exit chairs for example, they tend to clip through. That may never be resolved due to the free-form nature of the stuff we build and the infinite complexity of programming procedural body movements to avoid them. There is also the problem of characters getting stuck in built-up places. A layer of rigid character armour would either make these problems worse, or have to be allowed to clip through other solid items to avoid the collision detection work and regular snagging.

**Comment by ultrak2k** [comment]
> 
> 
> A thought about this idea is the complexity of how that armour would interact with ship interiors. When characters enter/exit chairs for example, they tend to clip through. That may never be resolved due to the free-form nature of the stuff we build and the infinite complexity of programming procedural body movements to avoid them. There is also the problem of characters getting stuck in built-up places. A layer of rigid character armour would either make these problems worse, or have to be allowed to clip through other solid items to avoid the collision detection work and regular snagging.

understandable, but do note that guns don't have collison, and are built out of blocks too

**Comment by ZachZent** [comment]
Here are my two cents on how I think armor should be added, though between mind and ultras either would be good.

I'd have a limited number of armor types (heavy, light, etc) each with various models and styles. Heavy armor has slow movement speed but higher hp and styles are more bulky like power armor, that sort of thing. With the armor, you can paint the base layer, secondary layer, detail layer sort of like how the loading screen is white, grey, orange respectively. You wouldn't be able to build off it, instead you can add all the stickers you want to personalize it. Maybe even can be without a grid and allow free placing and rotating. And you can copy the stickers you place onto other armors so your crew can have matching sets.

Since the armor is model based rather then fully customization, it can be replaced with future mods to get custom models. And you can still place all the stickers on it you want.

Internally, there are various slots you can place modules that change the nature/stats of the armor. Think Subnautica inventory with different sized modules. Your personal shield module would take up most of the slots, simple speed upgrade would only take up a few.

https://cdn.discordapp.com/attachments/271723356795961345/853057801634316318/unknown.png


Wouldn't call it better the ultras idea, just another way to do it


Edit: a slight deviation to the idea is not having different armor types but just one so everyone has the same number of modules that can be put in. Balanced and fair, though there should still be lots of styles to chose from like scifi space suit to fallout power armor

**Comment by ultrak2k** [comment]
> 
> 
> Here are my two cents on how I think armor should be added, though between mind and ultras either would be good.
> 
> I'd have a limited number of armor types (heavy, light, etc) each with various models and styles. Heavy armor has slow movement speed but higher hp and styles are more bulky like power armor, that sort of thing. With the armor, you can paint the base layer, secondary layer, detail layer sort of like how the loading screen is white, grey, orange respectively. You wouldn't be able to build off it, instead you can add all the stickers you want to personalize it. Maybe even can be without a grid and allow free placing and rotating. And you can copy the stickers you place onto other armors so your crew can have matching sets.
> 
> Since the armor is model based rather then fully customization, it can be replaced with future mods to get custom models. And you can still place all the stickers on it you want.
> 
> Internally, there are various slots you can place modules that change the nature/stats of the armor. Think Subnautica inventory with different sized modules. Your personal shield module would take up most of the slots, simple speed upgrade would only take up a few.
> 
> https://cdn.discordapp.com/attachments/271723356795961345/853057801634316318/unknown.png
> 
> Wouldn't call it better the ultras idea, just another way to do it

i also do fully support this idea, if brickbuilts aren't added

**Comment by ultrak2k** [comment]
> > Here are my two cents on how I think armor should be added, though between mind and ultras either would be good.
> > I'd have a limited number of armor types (heavy, light, etc) each with various models and styles. Heavy armor has slow movement speed but higher hp and styles are more bulky like power armor, that sort of thing. With the armor, you can paint the base layer, secondary layer, detail layer sort of like how the loading screen is white, grey, orange respectively. You wouldn't be able to build off it, instead you can add all the stickers you want to personalize it. Maybe even can be without a grid and allow free placing and rotating. And you can copy the stickers you place onto other armors so your crew can have matching sets.
> > Since the armor is model based rather then fully customization, it can be replaced with future mods to get custom models. And you can still place all the stickers on it you want.
> > Internally, there are various slots you can place modules that change the nature/stats of the armor. Think Subnautica inventory with different sized modules. Your personal shield module would take up most of the slots, simple speed upgrade would only take up a few.
> > https://cdn.discordapp.com/attachments/271723356795961345/853057801634316318/unknown.png
> > Wouldn't call it better the ultras idea, just another way to do it
> > Edit: a slight deviation to the idea is not having different armor types but just one so everyone has the same number of modules that can be put in. Balanced and fair, though there should still be lots of styles to chose from like scifi space suit to fallout power armor
> 
> What is someone doesn't like how heavy armor looks? How about adding those 3 armor types and allow players to personalise it's look? Avorion has this issue with heavy armor looking shitty, so players cover it with additional layers of light armor to make the ship look good, which has negative impact on perfomance

This discussion is about personal body armour not ship armour, I’m advocating for personal armour to be brick built so I can have a wild time and zach is going for a more pragmatic response

**Comment by Verum-Alex-Princeps** [comment]
Also customisable personal armor is good idea, but we already have many buggy issues with customisable handheld weapons, so I doubt that this feature for armor would work better. Instead I suggest to adding something like customisable armored exoskeleton which would work like mini-mech, so player will not be able to sit with it on ship's yoke and thus bugs with yoke won't happen

**Comment by Verum-Alex-Princeps** [comment]
> Here are my two cents on how I think armor should be added, though between mind and ultras either would be good.
> 
> I'd have a limited number of armor types (heavy, light, etc) each with various models and styles. Heavy armor has slow movement speed but higher hp and styles are more bulky like power armor, that sort of thing. With the armor, you can paint the base layer, secondary layer, detail layer sort of like how the loading screen is white, grey, orange respectively. You wouldn't be able to build off it, instead you can add all the stickers you want to personalize it. Maybe even can be without a grid and allow free placing and rotating. And you can copy the stickers you place onto other armors so your crew can have matching sets.
> 
> Since the armor is model based rather then fully customization, it can be replaced with future mods to get custom models. And you can still place all the stickers on it you want.
> 
> Internally, there are various slots you can place modules that change the nature/stats of the armor. Think Subnautica inventory with different sized modules. Your personal shield module would take up most of the slots, simple speed upgrade would only take up a few.
> 
> https://cdn.discordapp.com/attachments/271723356795961345/853057801634316318/unknown.png
> 
> Wouldn't call it better the ultras idea, just another way to do it
> 
> Edit: a slight deviation to the idea is not having different armor types but just one so everyone has the same number of modules that can be put in. Balanced and fair, though there should still be lots of styles to chose from like scifi space suit to fallout power armor

What is someone doesn't like how heavy armor looks? How about adding those 3 armor types and allow players to personalise it's look? Avorion has this issue with heavy armor looking shitty, so players cover it with additional layers of light armor to make the ship look good, which has negative impact on perfomance

**Comment by Verum-Alex-Princeps** [comment]
> > > Here are my two cents on how I think armor should be added, though between mind and ultras either would be good.
> > > I'd have a limited number of armor types (heavy, light, etc) each with various models and styles. Heavy armor has slow movement speed but higher hp and styles are more bulky like power armor, that sort of thing. With the armor, you can paint the base layer, secondary layer, detail layer sort of like how the loading screen is white, grey, orange respectively. You wouldn't be able to build off it, instead you can add all the stickers you want to personalize it. Maybe even can be without a grid and allow free placing and rotating. And you can copy the stickers you place onto other armors so your crew can have matching sets.
> > > Since the armor is model based rather then fully customization, it can be replaced with future mods to get custom models. And you can still place all the stickers on it you want.
> > > Internally, there are various slots you can place modules that change the nature/stats of the armor. Think Subnautica inventory with different sized modules. Your personal shield module would take up most of the slots, simple speed upgrade would only take up a few.
> > > https://cdn.discordapp.com/attachments/271723356795961345/853057801634316318/unknown.png
> > > Wouldn't call it better the ultras idea, just another way to do it
> > > Edit: a slight deviation to the idea is not having different armor types but just one so everyone has the same number of modules that can be put in. Balanced and fair, though there should still be lots of styles to chose from like scifi space suit to fallout power armor
> > 
> > 
> > What is someone doesn't like how heavy armor looks? How about adding those 3 armor types and allow players to personalise it's look? Avorion has this issue with heavy armor looking shitty, so players cover it with additional layers of light armor to make the ship look good, which has negative impact on perfomance
> 
> This discussion is about personal body armour not ship armour, I’m advocating for personal armour to be brick built so I can have a wild time and zach is going for a more pragmatic response

If it's about player armor, why you discussing the ship's armor as well here then?

**Comment by ultrak2k** [comment]
> > > > Here are my two cents on how I think armor should be added, though between mind and ultras either would be good.
> > > > I'd have a limited number of armor types (heavy, light, etc) each with various models and styles. Heavy armor has slow movement speed but higher hp and styles are more bulky like power armor, that sort of thing. With the armor, you can paint the base layer, secondary layer, detail layer sort of like how the loading screen is white, grey, orange respectively. You wouldn't be able to build off it, instead you can add all the stickers you want to personalize it. Maybe even can be without a grid and allow free placing and rotating. And you can copy the stickers you place onto other armors so your crew can have matching sets.
> > > > Since the armor is model based rather then fully customization, it can be replaced with future mods to get custom models. And you can still place all the stickers on it you want.
> > > > Internally, there are various slots you can place modules that change the nature/stats of the armor. Think Subnautica inventory with different sized modules. Your personal shield module would take up most of the slots, simple speed upgrade would only take up a few.
> > > > https://cdn.discordapp.com/attachments/271723356795961345/853057801634316318/unknown.png
> > > > Wouldn't call it better the ultras idea, just another way to do it
> > > > Edit: a slight deviation to the idea is not having different armor types but just one so everyone has the same number of modules that can be put in. Balanced and fair, though there should still be lots of styles to chose from like scifi space suit to fallout power armor
> > > 
> > > 
> > > What is someone doesn't like how heavy armor looks? How about adding those 3 armor types and allow players to personalise it's look? Avorion has this issue with heavy armor looking shitty, so players cover it with additional layers of light armor to make the ship look good, which has negative impact on perfomance
> > 
> > 
> > This discussion is about personal body armour not ship armour, I’m advocating for personal armour to be brick built so I can have a wild time and zach is going for a more pragmatic response
> 
> If it's about player armor, why you discussing the ship's armor as well here then?

huh where

**Comment by Verum-Alex-Princeps** [comment]
> > > > > Here are my two cents on how I think armor should be added, though between mind and ultras either would be good.
> > > > > I'd have a limited number of armor types (heavy, light, etc) each with various models and styles. Heavy armor has slow movement speed but higher hp and styles are more bulky like power armor, that sort of thing. With the armor, you can paint the base layer, secondary layer, detail layer sort of like how the loading screen is white, grey, orange respectively. You wouldn't be able to build off it, instead you can add all the stickers you want to personalize it. Maybe even can be without a grid and allow free placing and rotating. And you can copy the stickers you place onto other armors so your crew can have matching sets.
> > > > > Since the armor is model based rather then fully customization, it can be replaced with future mods to get custom models. And you can still place all the stickers on it you want.
> > > > > Internally, there are various slots you can place modules that change the nature/stats of the armor. Think Subnautica inventory with different sized modules. Your personal shield module would take up most of the slots, simple speed upgrade would only take up a few.
> > > > > https://cdn.discordapp.com/attachments/271723356795961345/853057801634316318/unknown.png
> > > > > Wouldn't call it better the ultras idea, just another way to do it
> > > > > Edit: a slight deviation to the idea is not having different armor types but just one so everyone has the same number of modules that can be put in. Balanced and fair, though there should still be lots of styles to chose from like scifi space suit to fallout power armor
> > > > 
> > > > 
> > > > What is someone doesn't like how heavy armor looks? How about adding those 3 armor types and allow players to personalise it's look? Avorion has this issue with heavy armor looking shitty, so players cover it with additional layers of light armor to make the ship look good, which has negative impact on perfomance
> > > 
> > > 
> > > This discussion is about personal body armour not ship armour, I’m advocating for personal armour to be brick built so I can have a wild time and zach is going for a more pragmatic response
> > 
> > 
> > If it's about player armor, why you discussing the ship's armor as well here then?
> 
> huh where

Ah, I guess, that I got confused and mistaken. Never thought that placing shield generator on player's armor suit is possible even in sci-fi

**Comment by ultrak2k** [comment]
> Also customisable personal armor is good idea, but we already have many buggy issues with customisable handheld weapons, so I doubt that this feature for armor would work better. Instead I suggest to adding something like customisable armored exoskeleton which would work like mini-mech, so player will not be able to sit with it on ship's yoke and thus bugs with yoke won't happen

isn’t that just mechs, especially with neovox - also I don’t really see “one thing is buggy, therefore the other thing is buggy” as fair because tsuna has been working on other things like the whole neovox deal


preferably brickbuilt armour for me would allow you to yes make armour but also make snazzy outfits lol, if exoskeletons can do the same somehow I wouldn’t be impartial even though I don’t think that’s really useful since at that point just use a mech

**Comment by ultrak2k** [comment]
also I don’t see the whole yoke thing as an unfixable issue because if it was I would be concerned on the playability of survival since you couldn’t really unstuck yourself as easily, and I don’t recall tsuna ever saying it’s an unsolvable issue and he tends to declare if something is really tricky to solvr

**Comment by Verum-Alex-Princeps** [comment]
> also I don’t see the whole yoke thing as an unfixable issue because if it was I would be concerned on the playability of survival since you couldn’t really unstuck yourself as easily, and I don’t recall tsuna ever saying it’s an unsolvable issue and he tends to declare if something is really tricky to solvr

well mechs are too big and can't be placed to a dropship for infantry for example, so I suggested minimech. Also the main question is HOW your brickbuilt armor will interact with ship's interrior and yoke:


> A thought about this idea is the complexity of how that armour would interact with ship interiors. When characters enter/exit chairs for example, they tend to clip through. That may never be resolved due to the free-form nature of the stuff we build and the infinite complexity of programming procedural body movements to avoid them. There is also the problem of characters getting stuck in built-up places. A layer of rigid character armour would either make these problems worse, or have to be allowed to clip through other solid items to avoid the collision detection work and regular snagging.

**Comment by ultrak2k** [comment]
ah yeah true, I guess mechs are rather large, maybe even with 1/8 bricks, to be fair though I do remember in the OG trailer one coming out of a somewhat small ship - also interactions? Haven’t a clue honestly, but I’m sure there’s a solution to it

> > also I don’t see the whole yoke thing as an unfixable issue because if it was I would be concerned on the playability of survival since you couldn’t really unstuck yourself as easily, and I don’t recall tsuna ever saying it’s an unsolvable issue and he tends to declare if something is really tricky to solvr
> 
> well mechs are too big and can't be placed to a dropship for infantry for example, so I suggested minimech. Also the main question is HOW your brickbuilt armor will interact with ship's interrior and yoke:
> 
> > A thought about this idea is the complexity of how that armour would interact with ship interiors. When characters enter/exit chairs for example, they tend to clip through. That may never be resolved due to the free-form nature of the stuff we build and the infinite complexity of programming procedural body movements to avoid them. There is also the problem of characters getting stuck in built-up places. A layer of rigid character armour would either make these problems worse, or have to be allowed to clip through other solid items to avoid the collision detection work and regular snagging.

**Comment by ultrak2k** [comment]
perhaps armour doesn’t need to clip at all? depends on how it scales

**Comment by ultrak2k** [comment]
it’s a tough question but I honestly just threw it out there as an extrapolation of brick guns and neovox

**Comment by ZachZent** [comment]
> What is someone doesn't like how heavy armor looks? How about adding those 3 armor types and allow players to personalise it's look? Avorion has this issue with heavy armor looking shitty, so players cover it with additional layers of light armor to make the ship look good, which has negative impact on perfomance

I did say that there would be various styles of armor in each armor type. So you have a cleaner heavy armor and a more unclean one, but heavy is heavy so it will be generally bulkier.

Or with the second option of no light/heavy and everyone gets the same "armor" in a gameplay sense (same stats, same number of module spots), all the armor options are open from space suit to power armor as it would all be the same except visually. Plus it is even better for modders as they can add any armor they want like ultra's maid armor and no one would have to determine if say the skimpy maid armor actually is heavy. Bit less confusion for a bit less variability

**Comment by Garrett-C** [comment]
I would agree with zach that an item-based system should be the way to go. 

In my opinion, a built armour just sounds too similar to mechs. 

An armour system that let's you choose between a couple of tiers and aesthetics sounds like the best way to go. This also could be relatively easy to mod so if you want to add old fashioned scuba gear as an armour you can, if you want to add halo armour you can, if you want to add a pair of swim shorts and a tshirt you can. Obviously, these would be mods and not the core armours included in the game.

On top of this customisation like having the armour be made up of a couple of core pieces could allow for more progression in gameplay and more customisation. I would suggest at a minimum head, chest, legs, boots and under armour with some accessory slots. These accessory slots could work for practical things like O2 or a jetpack but could also work for purely aesthetic things like a hat or some gloves etc. A nice amount of these accessory slots could allow for a decent level of character customisation with base game accessories and opens up a whole load of possibilities when modders start messing with it.

**Comment by cloudrambler** [comment]
I think that the logical solution for player based customisations, such as armour, is to use the closest style and method to the characters themselves. Characters are soft and flexible in the way they are animated, so an armour design that could use the same design rules would be the most elegant, from a game dev POV.

Also I think that the visual contrast between block based armour and the characters would be awkward. STEVO was inspired by Minecraft, but has already well surpassed that level of aesthetic subtlety. Do we want to re-introduce more visual chunkiness, or do we want to make the most of the style that we have got?

We also have to think about how the game will be fleshed out and what aspects are valuable. Is STEVO going to be a character RPG or is the main focus that we can make detailed ships to work with. Yes, we can be the character and explore within the ships etc., but I still think the main thrust of the game is about what we do with those ships, not what we do independently of them.

Some may not agree with me, but I think there are plenty of games that let you explore space environments as a badass hero or other, but there are no other games that let you explore the universe in something that feels like your home and which you actually care about because you properly built it.

The game will feel flat without character customisation, both visual and technical, but that aspect should be limited to avoid compromising the main interaction with the custom ship interiors.

**Comment by ultrak2k** [comment]
Dayum this is all pretty interesting, wasnt expecting this much discussion from me asking “can I please be able to have maid outfits in the game” disguised as an armour building question
